lessons/advanced-displays/langs/en-us/index.adoc

```diff
@@ -0,0 +1,217 @@
+- Open the @starter-file{custom-scatter-plot}.
+- With a partner or in small groups, complete @printable-exercise{image-scatter-explore.adoc}.
+
+
+The contract for `image-scatter-plot` looks pretty different from other contracts we've seen.
+
+[.forceShading]
+--
+```
+image-scatter-plot :: (
+  t  :: Table,
+  xs :: String,
+  ys :: String,
+  f  :: (Row -> Image)
+) -> Image
+```
+--
+
+The function `image-scatter-plot` has an interesting Domain: _Table, String, String and... what is that last part?... *a Function that consumes a Row and produces an Image*!_
+
+=== Synthesize
+
+* How does having the dots labeled with the species change your understanding of the data?
+
+__Students should recognize that the `species-tag` @vocab{scatter plot} makes it clear that we may want to analyze each species separately, rather than grouping them all together. (In the Grouped Samples lesson, students will learn how to do just that!)__
+
+@center{@image{images/species-tag.png, "A scatter plot with species names for each animal in place of a small, uniform dot. The images are generated by the species-tag function", 450}}
+
+* What other ways could we use the columns in each row to pack more information into our scatter plots?
+** Use legs and age to determine width and height of a rectangle
+** Use different colors or shapes based on species, sex, or fixed-status
+
+== Piecewise/Conditional Functions @duration{15 minutes}
+
+=== Overview
+
+Students discover how to use @vocab{conditionals} - @vocab{piecewise functions} in math - to create more complex scatter plots, and also custom bar charts, pie charts, and histograms. Experimentation becomes the motivation for more practice with Row-consuming functons.
+
+*NOTE:* Math teachers may want their students to confront piecewise functions more deeply, and CS teachers may want to spend more time on conditionals. While not a part of the Data Science pathway, the @lesson-link{piecewise-functions-conditionals/} lesson includes a lot of supporting material and practice pages for these topics, with greater emphasis on the math connections.
+
+=== Launch
+So far, we've seen that...
+
+* the `scatter-plot` function makes uniform blue dots
+* the `image-scatter-plot` function can label each point with a custom image, computed by a helper function that consumes Rows and produces Images.
+
+How could we use _different color dots_ for each species? Or perhaps draw different shapes depending on whether an animal was fixed or not?
+
+This requires a more powerful kind of function, known in math as a @vocab{piecewise function} because it has more than one "piece". In programming, these are often called "conditional" functions, because each piece is used depending on which condition is met.
+
+=== Investigate
+
+[.lesson-instruction]
+* __Starting on paper__, complete @printable-exercise{pages/piecewise-explore.adoc}.
+* When you've finishing thinking and writing, open the @starter-file{piecewise-displays} and experiment!.
```

Comment:
I don't get that error -- I got a _different one_ (now fixed), but I want to make sure I'm looking at the right file.

Added directive.

Reply:
No longer getting an error. And, wow, this is neat!

Reply:
Now that I am able to proceed with piecewise-explore, I have additional (minor) feedback, which is that question 3 needs some improvement. You say, "What do these two displays show, that we didn't see before without piecewise functions?" Kids see *four* displays when they hit run. I think you might want the following questions instead of Q3:

- Compare the first display (scatter-plot) with the third display (image-scatter-plot). What can you see now that you couldn't see before?
- Compare the second display (histogram) with the fourth display (image-histogram). What can you see now that you couldn't see before?
- Can you glean any new insights by looking at image-scatter-plot and image-histogram _together_?

Reply:
@schanzer - seeing that this comment is marked as "outdated" so pinging you to make sure it doesn't get missed.

Reply:
Oooh, thanks for flagging this! I love these suggestions, and have implemented all of them. Resolve if you're happy?